### DISCUSSION TITLE
SCD2 with Hash without key/date columns

### DISCUSSION THREAD

**Discussion by SuvarnaYZetaris** [osalvador/ReplicaDB#124] [General]
Can we do SCD2 with replicadb also with hash if we don't have key/date columns? I tried with debezium however its not supporting other JDBC connector.

**Comment by osalvador** [comment]
Hi @SuvarnaYZetaris 

If you don't have any date or incremental key column, the are two solutions using ReplicaDB:
- Replicate the entire table on each replication.
- Create a trigger on the source table with the ON INSERT OR UPDATE statement and copy the row to another table. Subsequently, replicate this table with ReplicaDB with complete mode and with the `sink.disable.truncate=true` parameter. 

Regards!

**Reply by SuvarnaYZetaris** [reply]
Thanks for Quick Reply Oscar, is there any way (some custom script) to keep modified row in same sink table & maintain flag as retired?
